src/groovy/org/transmartproject/db/biomarker/AssocBioMarkerPropertiesConstraint.groovy

```diff
@@ -0,0 +1,39 @@
+package org.transmartproject.db.biomarker
+
+import grails.orm.HibernateCriteriaBuilder
+import org.hibernate.criterion.DetachedCriteria
+import org.hibernate.criterion.Property
+import org.hibernate.criterion.Restrictions
+import org.transmartproject.core.exceptions.InvalidArgumentsException
+
+class AssocBioMarkerPropertiesConstraint implements BioMarkerCriteriaConstraint {
+
+    Map<String, Object> parameters
+
+    @Override
+    void doWithCriteriaBuilder(HibernateCriteriaBuilder criteria) {
+        if (!parameters) {
+            throw new InvalidArgumentsException('No parameters are specified.')
+        }
+
+        DetachedCriteria detachedCriteria = HibernateCriteriaBuilder.getHibernateDetachedCriteria(
+                BioDataCorrelationCoreDb.where {
+                    projections {
+                        property('leftBioMarker.id')
+                    }
+                }
+        )
+
+        def assocBioMarkerSubCriteria = detachedCriteria.createCriteria('rightBioMarker')
+        parameters.each { entry ->
+            assocBioMarkerSubCriteria.add(
+                    entry.value instanceof Collection ?
+                            Restrictions.in(entry.key, entry.value)
+                            : Restrictions.eq(entry.key, entry.value)
+            )
+        }
+
+        criteria.add(Property.forName('id').in(detachedCriteria))
+    }
```

Comment:
Why? The correlations all follow the same model with only the table and the name of one column differing.